runtime/src/serde_snapshot/tests.rs

```diff
@@ -214,6 +218,47 @@ fn test_bank_serialize_style(serde_style: SerdeStyle) {
     )
     .unwrap();
 
+    let accounts_hash = if update_accounts_hash {
```

Comment:
@brooksprumo here's a test. I understand more will change based on slack discussions, but this is at least the framework of a test that should exercise what we need.